javascript/code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,401 @@
+"use strict";
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function that iterates over an array of people objects
+and creates a new list of each person's full name using the array method 'map'.
+Each object will have the shape {firstName:string, lastName:string}
+E.g. [ { firstName:"Jane", lastName:"Doe" }, { firstName:"James", lastName:"Bond"}]
+should convert to ["Jane Doe", "James Bond"]
+Note the space in between first and last names.
+You can assume that neither firstName nor lastName will be blank
+------------------------------------------------------------------------------------------------ */
+const toLastNames = (people) => {
+  return people.map((obj) => obj.firstName + " " + obj.lastName);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named addValues that, given an array of numbers as input, uses reduce to add the values in the array.
+
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr) => {
+  let sum = arr.reduce((acc, curr) => {
+    acc += curr;
+    return acc;
+  }, 0);
+  return sum;
```

Comment:
To tighten this up, you could always just return arr.reduce, instead of putting it in a variable first. But What you're doing isn't wrong. This absolutely works if it helps you think about it in a way that makes sense to you.